libs/remote_storage/src/local_fs.rs

```diff
@@ -171,23 +162,19 @@ impl RemoteStorage for LocalFs {
             }
         }
 
-        // Note that PathBuf starts_with only considers full path segments, but
+        // Note that Utf8PathBuf starts_with only considers full path segments, but
         // object prefixes are arbitrary strings, so we need the strings for doing
         // starts_with later.
-        let prefix = full_path.to_string_lossy();
+        let prefix = full_path.to_string();
 
         let mut files = vec![];
         let mut directory_queue = vec![initial_dir.clone()];
         while let Some(cur_folder) = directory_queue.pop() {
-            let mut entries = fs::read_dir(cur_folder.clone()).await?;
-            while let Some(entry) = entries.next_entry().await? {
-                let file_name: PathBuf = entry.file_name().into();
-                let full_file_name = cur_folder.clone().join(&file_name);
-                if full_file_name
-                    .to_str()
-                    .map(|s| s.starts_with(prefix.as_ref()))
-                    .unwrap_or(false)
-                {
+            let mut entries = cur_folder.read_dir_utf8()?;
+            while let Some(Ok(entry)) = entries.next() {
+                let file_name = entry.file_name();
+                let full_file_name = cur_folder.clone().join(file_name);
```

Comment:
```suggestion
                let full_file_name = cur_folder.join(file_name);
```

I would expect this clone to be unnecessary, unless you just removed it in the previous push, because `Path::join` creates a new pathbuf, it seems it's similar to in camino.

Reply:
Done https://github.com/neondatabase/neon/pull/5352/commits/b437516edf8ae9e85cc20ae78b77fb8ee003a981